javalib/src/main/scala/java/util/regex/Matcher.scala

```diff
@@ -85,8 +85,8 @@ final class Matcher private[regex] (var _pattern: Pattern,
         var i = 0
         while (i < nMatches) {
           val m     = matches + i
-          val start = (m.data - in).toInt
-          val end   = start + m.length
+          val start = if (m.data == null) -1 else (m.data - in).toInt
+          val end   = if (m.data == null) -1 else start + m.length
```

Comment:
This seems to be like a workaround that doesn't address the root cause of the issue: we add `+ 1` to the number of groups in https://github.com/scala-native/scala-native/pull/740/files#diff-ef6e5cb9d8056c08c6441a0d16a64425R39, and then when we try to read the data of that last group it's simply not there (i.e. `m.data == null`.) Can we fix the root cause instead?